File: C#/0001. Two Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int,int> numsDic=new Dictionary<int,int>();
        for(int i=0;i<nums.Length;i++){
            if (numsDic.ContainsKey(target-nums[i])){
                return new int[]{numsDic[target-nums[i]],i};
            }
            numsDic[nums[i]]=i;
        }
        return new int[0];
    }
}

File: archives/20190519python/0001. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dictnums={}
        i=0
        while(i<len(nums)):
            if(target-nums[i] not in dictnums):
                dictnums[nums[i]]=i
            else:
                return[dictnums[target-nums[i]],i]
            i+=1
            
            
#==================================================================================
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dictnums={}
        for i,n in enumerate(nums):
            if target-n in dictnums:
                return [dictnums[target-n],i]
            dictnums[n]=i
